[GStreamer][WebRTC] Implement GstMappedRtpBuffer

philn opened the pull request:
#### 7878d72b0fd7c5c84e185733ffc2d6ed65bccc02
<pre>
[GStreamer][WebRTC] Implement GstMappedRtpBuffer
<a href="https://bugs.webkit.org/show_bug.cgi?id=251428">https://bugs.webkit.org/show_bug.cgi?id=251428</a>

Reviewed by Xabier Rodriguez-Calvar.

With this new RAII object for mapping RTP buffers we don&apos;t need to manually unmap anymore.

Canonical link: <a href="https://commits.webkit.org/261074@main">https://commits.webkit.org/261074@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ffe5ca34ed2ec179e12e4b543e5ba7ac0183860

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110490 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19574 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43080 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1859 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119378 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114437 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21004 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10722 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102725 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116232 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15642 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/98827 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43878 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97604 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30499 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85745 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12228 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31834 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12805 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/8799 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18168 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51454 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7673 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14672 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

calvaris reviewed (changes requested):
If this implementation is the same as the `GstMappedBuffer` with the only difference of map/unmap functions, I think we should do some templating here and optionally pass the map/unmap functions. Then, if you want, you can create the `GstRTPMappedBuffer` with a `using`.

shivamidow commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.h:
```diff
@@ -268,5 +270,64 @@ WARN_UNUSED_RETURN GRefPtr<GstCaps> capsFromRtpCapabilities(const RTCRtpCapabili
+
+    void unmapEarly()
+    {
+        if (m_isValid) {
```
* Unofficial review *
Nit. We may want an early return instead.

calvaris reviewed (approved):
B E A U T I F U L !

philn committed:
[GStreamer][WebRTC] Implement GstMappedRtpBuffer
https://bugs.webkit.org/show_bug.cgi?id=251428

Reviewed by Xabier Rodriguez-Calvar.

With this new RAII object for mapping RTP buffers we don't need to manually unmap anymore.

Canonical link: https://commits.webkit.org/261074@main
